libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -277,7 +277,7 @@ bool ReferencesResolver::visit(InlineAssembly const& _inlineAssembly)
 	// Will be re-generated later with correct information
 	// We use the latest EVM version because we will re-run it anyway.
 	assembly::AsmAnalysisInfo analysisInfo;
-	boost::optional<Error::Type> errorTypeForLoose = m_experimental050Mode ? Error::Type::SyntaxError : Error::Type::Warning;
+	boost::optional<Error::Type> errorTypeForLoose = Error::Type::SyntaxError;
```

Comment:
Either keep this and in `TypeChecker` or merge into the `AsmAnalyzer` line in both cases. While first I was going to suggest not to keep the variable, at least it helps documenting the code.

Well, actually, as a second PR the parameter to `AsmAnalyzer` could be removed entirely.

Reply:
@axic I thought we still might want to keep loose style for the `solc --assemble` mode, but we can also fully remove it.

Reply:
@axic I'd even go that far in removing that ctor param alltogether (no need for a seperate PR, IMHO), what do you think? /cc @chriseth

Reply:
> @axic I thought we still might want to keep loose style for the solc --assemble mode, but we can also fully remove it.

Oh right, for that reason I would not change it too much in this PR then. If we decode to remove that, should be in a separate PR.